server/src/main/scala/org/http4s/server/package.scala

```diff
@@ -1,13 +1,44 @@
 package org.http4s
 
-import scalaz.{OptionT, Kleisli}
+import scalaz.{Equal, Kleisli, Monoid}
 import scalaz.concurrent.Task
-import scalaz.std.option._
 import scalaz.syntax.kleisli._
-import scalaz.syntax.monad._
-import scalaz.syntax.traverse._
+
+package server {
+
+  /**
+    * Encapsulates the notion of fallthrough orElse for a Service
+    * For any given B, if a Fallthrough[B] exists within implicit context
+    * then [[Service#orElse]] can be used.
+    */
+  trait Fallthrough[B] {
+    def isFallthrough(a: B): Boolean
+    def fallthrough[A](fst: B, snd: Service[A, B]): Service[A, B] =
```

Comment:
A `Service` needs to know about `Fallthrough`, but `Fallthrough` doesn't need to know anything about `Service`.  Same package, so not a real problematic cyclic dependency... I just expected this to live elsewhere.